### PR TITLE
pass locale to ZAS Package

### PR DESCRIPTION
**Pull request by craig-day**
Use the locale parameter sent by the classic apps helper. 

/cc @pdeuter @benhass @princemaple 

### References
 - Classic Change: https://github.com/zendesk/zendesk/pull/16964

### Risks
 - Low. Just adding a key into the parameters hash.

### CODE REVIEW

File: lib/zendesk_apps_tools/server.rb [hunk 1]
```diff
@@ -17,8 +17,13 @@ class Server < Sinatra::Base
           last_mtime = curr_mtime
         end
       end
+      settings.parameters.merge!( { locale: normalize_locale(params['locale']) } )
```
Comment (princemaple) [inline lib/zendesk_apps_tools/server.rb]:
```ruby
settings.parameters.merge!(locale: normalize_locale(params['locale']))
```

**Comment by princemaple** [comment]
:+1: otherwise

File: lib/zendesk_apps_tools/server.rb [hunk 2]
```diff
@@ -17,8 +17,13 @@ class Server < Sinatra::Base
...
 
       ZendeskAppsSupport::Package.new(settings.root).readified_js(nil, settings.app_id, "http://localhost:#{settings.port}/", settings.parameters)
     end
+
+    def normalize_locale(locale)
```
Comment (liulikun) [inline lib/zendesk_apps_tools/server.rb]: `locale` could be nil? Can set default here `def normalize_locale(locale = 'en')`.

**Comment by liulikun** [comment]
:+1:

**Comment by craig-day** [comment]
@princemaple @liulikun Decided to not touch the locale in here and just pass `params['locale']` to ZAS.  I am going to change the logic there a little bit to handle the situation more appropriately.

**Comment by craig-day** [comment]
hey @princemaple what is the version bumping process for this? I would like to bump before and after this merge so its easy to not use if problems come up.

File: lib/zendesk_apps_tools/server.rb [hunk 3]
```diff
@@ -17,6 +17,7 @@ class Server < Sinatra::Base
           last_mtime = curr_mtime
         end
       end
+      settings.parameters.merge!(locale: params['locale'])
```
Comment (liulikun) [inline lib/zendesk_apps_tools/server.rb]: An edge case would be if the app also names a parameter 'locale', then we're going to overwrite it here. Can we pass it through without touching parameters?

**Comment by princemaple** [comment]
Why would you need to bump twice?

**Comment by craig-day** [comment]
it was never bumped after the pseudo translate code was added in. I don't really need to, just a though.

File: lib/zendesk_apps_tools/server.rb (hunk 3 above)
Comment (craig-day) [inline lib/zendesk_apps_tools/server.rb]: @liulikun we could add an optional parameter to the end of the method signature for `ZendeskAppsSupport::Package#initialize` if that seems better?

**Comment by princemaple** [comment]
I see. Not sure if you have push rights.
I normally just do
```
rake bump:patch
rake release
```

**Comment by princemaple** [comment]
If you don't have push rights, I can do it for you.

File: lib/zendesk_apps_tools/server.rb (hunk 3 above)
Comment (pdeuter) [inline lib/zendesk_apps_tools/server.rb]: Shouldn't locale be a standard parameter for *all* apps.  I think it should be.  I can hardly imagine that an app would have a locale parameter that did not have the same semantics as this locale parameter.

Comment (craig-day) [inline lib/zendesk_apps_tools/server.rb]: @pdeuter I think he is pointing out the app parameter `locale` could potentially be overwritten by the URL parameter being sent along from a classic helper.  I'm not sure if this is the same issue you are referring to.

File: lib/zendesk_apps_tools/server.rb [hunk 4]
```diff
@@ -18,7 +18,7 @@ class Server < Sinatra::Base
         end
       end
 
-      ZendeskAppsSupport::Package.new(settings.root).readified_js(nil, settings.app_id, "http://localhost:#{settings.port}/", settings.parameters)
+      ZendeskAppsSupport::Package.new(settings.root).readified_js(nil, settings.app_id, "http://localhost:#{settings.port}/", settings.parameters, params['locale'])
```
Comment (craig-day) [inline lib/zendesk_apps_tools/server.rb]: @princemaple @liulikun how do we feel about doing it this way instead? Corresponding change is [here](https://github.com/zendesk/zendesk_apps_support/pull/83/files#diff-f9dc5da4f07a7cd59a608b6da769f0ecR97)

Comment (liulikun) [inline lib/zendesk_apps_tools/server.rb]: I like this approach better.

**Comment by craig-day** [comment]
@princemaple @liulikun bump.

**Comment by liulikun** [comment]
:+1:

**Comment by craig-day** [comment]
@princemaple @liulikun merged and bumped. Can someone release the new version? Thanks!

**Comment by princemaple** [comment]
Released

**Pull request merged**